### PR TITLE
Update RazorSourceGenerator.cs

### PR DESCRIPTION
**Pull request by captainsafia**

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]
I made a couple additional changes. Please have a look.

**Comment by captainsafia** [comment]
@pranavkm Changes look sensible.

**Comment by pranavkm** [comment]
I think this needs a newer targeting pack: 

```
C:\h\w\AA4F091C\t\dotnetSdkTests\3iw4y05e.t0j\UpToDateReloa---143E05B3\Microsoft.NET.Sdk.Razor.SourceGenerators\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator\SimpleMvc.UnifiedAssembly.Info.cs(1,158): error CS0234: The type or namespace name 'ConsolidatedAssemblyApplicationPartFactory' does not exist in the namespace 'Microsoft.AspNetCore.Mvc.ApplicationParts' (are you missing an assembly reference?) [C:\h\w\AA4F091C\t\dotnetSdkTests\3iw4y05e.t0j\UpToDateReloa---143E05B3\SimpleMvc.csproj]
```

File: src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs [hunk 1]
```diff
@@ -252,11 +254,11 @@ private static IReadOnlyList<TagHelperDescriptor> GetTagHelperDescriptorsFromRef
             return tagHelperDescriptors;
         }
 
-        private static void PopulateAssemblyInfo(GeneratorExecutionContext context)
+        private static SourceText GetProvideApplicationPartFactorySourceText()
         {
             var typeInfo = "typeof(global::Microsoft.AspNetCore.Mvc.ApplicationParts.ConsolidatedAssemblyApplicationPartFactory)";
```
Comment (pranavkm) [inline src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs:259]:
```suggestion
            var typeInfo = "Microsoft.AspNetCore.Mvc.ApplicationParts.ConsolidatedAssemblyApplicationPartFactory, Microsoft.AspNetCore.Mvc.Razor";
```

File: src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs [hunk 2]
```diff
@@ -252,11 +254,11 @@ private static IReadOnlyList<TagHelperDescriptor> GetTagHelperDescriptorsFromRef
...
-            var assemblyInfo = $@"[assembly: global::Microsoft.AspNetCore.Mvc.ApplicationParts.ProvideApplicationPartFactoryAttribute(""{typeInfo}"")]";
-            context.AddSource($"{context.Compilation.AssemblyName}.UnifiedAssembly.Info", SourceText.From(assemblyInfo, Encoding.UTF8));
+            var assemblyInfo = $@"[assembly: global::Microsoft.AspNetCore.Mvc.ApplicationParts.ProvideApplicationPartFactoryAttribute({typeInfo})]";
```
Comment (pranavkm) [inline src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs:260]:
```suggestion
            var assemblyInfo = $@"[assembly: global::Microsoft.AspNetCore.Mvc.ApplicationParts.ProvideApplicationPartFactoryAttribute(""{typeInfo}"")]";
```

**Review by pranavkm — COMMENTED** [review]
Might work

File: src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs (hunk 1 above)
Comment (captainsafia) [reply src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs]: Yeah, I tried this in ba01d88 but it didn't work. I missed the need for `Microsoft.AspNetCore.Mvc.Razor`.

**Pull request merged**